src/test/java/com/direwolf20/buildinggadgets/building/placementTests/SingleTypeProviderTest.java

```diff
@@ -0,0 +1,48 @@
+package com.direwolf20.buildinggadgets.building.placementTests;
+
+import com.direwolf20.buildinggadgets.common.building.placement.SingleTypeProvider;
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import org.junit.jupiter.api.*;
+
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Disabled("requires custom runner with minecraft started")
+public class SingleTypeProviderTest {
+
+    //TODO implement custom runner and add tests
+    //this will not work since initializing a block instance requires sound engine and texture engine
+    private final IBlockState state = new Block(Material.AIR).getDefaultState();
```

Comment:
This is what happens when you trying to write grouped test cases over a week...
Also the serialization/deserialization still won't work since `NBTUtil.readBlockState(NBTTagCompound)` searches through block registry.